agents/rust/mediator/mediation/.github/workflows/rust-basic.yml

```diff
@@ -0,0 +1,130 @@
+name: Rust basic workflow
+
+on:
+  push:
+    branches: [ "main", "ci-integration-test", "ci-cd", "dev", aries-vcx-mediator-integration]
+  pull_request:
+    branches: [ "main", "ci-integration-test", "ci-cd", "dev", aries-vcx-mediator-integration]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check:
+    name: Check for obvious errors
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    - name: Cargo cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-cargo-check-cache-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-cargo-
+    - name: cargo check
+      run: cargo check --verbose
+  build:
+    needs: check
+    name: Build and cache
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Cargo build cache
+      # Restore / save build cache to speed up operations
+      # If no cache was found and job succeeds cache of paths will be created
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-cargo-build-cache-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-cargo-    
+    - name: Build
```

Comment:
It would be more valuable to build docker images for mediator, then publishing the binary itself. 

But hey, why not. Though note the usefulness of building this binary is limited, as you are only building for a single architecture.